SwiftVoiceQuickstart/ViewController.swift

```diff
@@ -261,32 +261,30 @@ extension ViewController: PushKitEventDelegate {
     func credentialsUpdated(credentials: PKPushCredentials) {
         guard let accessToken = fetchAccessToken() else { return }
         
-        let deviceToken = credentials.token.map { String(format: "%02x", $0) }.joined()
-        
-        let userDefaults = UserDefaults.standard
-        let cachedDeviceToken = userDefaults.string(forKey: kCachedDeviceToken) ?? deviceToken
-        
-        if cachedDeviceToken != deviceToken {
-            /*
-             * Perform registration if a new device token is detected.
-             */
-            TwilioVoice.register(withAccessToken: accessToken, deviceToken: cachedDeviceToken) { error in
-                if let error = error {
-                    NSLog("An error occurred while registering: \(error.localizedDescription)")
-                } else {
-                    NSLog("Successfully registered for VoIP push notifications.")
-                    
-                    /*
-                     * Save the device token after successfully registered.
-                     */
-                    userDefaults.set(cachedDeviceToken, forKey: kCachedDeviceToken)
-                }
+        if let cachedDeviceToken = UserDefaults.standard.data(forKey: kCachedDeviceToken), cachedDeviceToken == credentials.token {
+            return
+        }
```

Comment:
I'm pretty sure this will work. I think `==` will just work even with the optional.
```swift
guard 
    let accessToken = fetchAccessToken(),
    UserDefaults.standard.data(forKey: kCachedDeviceToken) != credentials.token
    else {
        return
}
```
Is that what you were looking for? 

Or is it the data comparison that you are asking about? Is that data UTF8 encoded string I assume?

Style for multiline guard is interesting. I use from here: https://github.com/raywenderlich/swift-style-guide#golden-path

Reply:
I would normally decode to string and compare 2 strings. Am I missing something with that `.data`?

Reply:
Thanks @timrozum. The update in the new version of SDK takes the `data` directly in case Apple decided to change the format or the length of the device token in the future. This way we only have to patch the SDK instead of having to update our examples to show developers how to decode the data to string.

Reply:
Oh makes sense. Looks good!